pkgs/applications/editors/rstudio/use-system-yaml-cpp.patch

```diff
@@ -0,0 +1,34 @@
+diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
+index 30493c7a06..b47f04f84d 100644
+--- a/src/cpp/CMakeLists.txt
++++ b/src/cpp/CMakeLists.txt
+@@ -197,7 +197,12 @@ endif()
+ # yaml-cpp
+ set(YAML_CPP_VERSION "0.8.0")
+ set(YAML_CPP_REVISION "f7320141120f720aecc4c32be25586e7da9eb978")
+-if(CMAKE_VERSION VERSION_GREATER "3.14")
++if(RSTUDIO_USE_SYSTEM_YAML_CPP)
++   # NOTE: defines the following CMake variables:
++   # - YAML_CPP_INCLUDE_DIR
++   # - YAML_CPP_LIBRARIES
++   find_package(yaml-cpp REQUIRED)
++elseif(CMAKE_VERSION VERSION_GREATER "3.14")
+ 
+    if(WIN32)
+       add_definitions(-DYAML_CPP_STATIC_DEFINE)
+@@ -213,14 +218,9 @@ if(CMAKE_VERSION VERSION_GREATER "3.14")
+    set(YAML_CPP_INCLUDE_DIR "${yaml-cpp_SOURCE_DIR}/include")
+    set(YAML_CPP_LIBRARIES   yaml-cpp::yaml-cpp)
+ 
+-elseif(NOT RSTUDIO_USE_SYSTEM_YAML_CPP)
++else()
+    set(YAML_CPP_INCLUDE_DIR "${RSTUDIO_TOOLS_ROOT}/yaml-cpp/${YAML_CPP_VERSION}/include")
+    set(YAML_CPP_LIBRARIES   "${RSTUDIO_TOOLS_ROOT}/yaml-cpp/${YAML_CPP_VERSION}/build/libyaml-cpp.a")
+-else()
+-   # NOTE: defines the following CMake variables:
+-   # - YAML_CPP_INCLUDE_DIR
+-   # - YAML_CPP_LIBRARIES
+-   find_package(yaml-cpp REQUIRED)
+ endif()
+ 
+ if(yaml-cpp_FOUND AND NOT YAML_CPP_LIBRARIES)
```

Comment:
Fetch the patch from the rstudio repository with `fetchPatch` instead of checking the content into this repo.

Reply:
thanks, and done!